REGRESSION(296337@main): currentcolor used in custom property no longer resolves correctly

weinig opened the pull request:
#### 6496fbaca29d34171082af8e739f6b36e8c6522a
<pre>
REGRESSION(296337@main): currentcolor used in custom property no longer resolves correctly
<a href="https://bugs.webkit.org/show_bug.cgi?id=296134">https://bugs.webkit.org/show_bug.cgi?id=296134</a>
<a href="https://rdar.apple.com/156076502">rdar://156076502</a>

Reviewed by Tim Nguyen.

Performs a partial revert of 296337@main, maintaining the CSS side of things,
but adding back the secondary serialization of the color Style types. To make
things work, the new secondary serialization functions all needed a new name.
`serializationForCSSTokenization` was chosen to clarify these are really only
for the tokenization needs.

* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-006-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-006.html: Added.
* Source/WebCore/style/StyleCustomProperty.cpp:
* Source/WebCore/style/StyleCustomProperty.h:
* Source/WebCore/style/values/color/StyleColor.cpp:
* Source/WebCore/style/values/color/StyleColor.h:
* Source/WebCore/style/values/color/StyleColorLayers.cpp:
* Source/WebCore/style/values/color/StyleColorLayers.h:
* Source/WebCore/style/values/color/StyleColorMix.cpp:
* Source/WebCore/style/values/color/StyleColorMix.h:
* Source/WebCore/style/values/color/StyleContrastColor.cpp:
* Source/WebCore/style/values/color/StyleContrastColor.h:
* Source/WebCore/style/values/color/StyleCurrentColor.cpp:
* Source/WebCore/style/values/color/StyleCurrentColor.h:
* Source/WebCore/style/values/color/StyleRelativeColor.h:
* Source/WebCore/style/values/color/StyleResolvedColor.cpp:
* Source/WebCore/style/values/color/StyleResolvedColor.h:

Canonical link: <a href="https://commits.webkit.org/297600@main">https://commits.webkit.org/297600@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/756be6780f7dbfb3b8fd1d707d9423526a9e926e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112132 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31805 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22298 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118159 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62604 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114094 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40384 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85195 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/35980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115079 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25951 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100897 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65624 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25258 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19036 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62051 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95337 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19112 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121490 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39170 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29164 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94018 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39551 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97140 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93841 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24016 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39060 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16847 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/35234 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39063 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44582 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38698 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42064 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40414 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-006.html:
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<link rel="help" href="https://drafts.csswg.org/css-properties-values-api/#calculation-of-computed-values" />
+<link rel="author" title="Sam Weinig" href="sam@webkit.org">
+<link rel="match" href="registered-property-computation-color-001-ref.html">
```
```suggestion
<link rel="match" href="../reference/ref-filled-green-100px-square-only.html">
```

weinig replied:
That file doesn't seem to exist and mine matches the other tests in the directory.

weinig committed:
REGRESSION(296337@main): currentcolor used in custom property no longer resolves correctly
https://bugs.webkit.org/show_bug.cgi?id=296134
rdar://156076502

Reviewed by Tim Nguyen.

Performs a partial revert of 296337@main, maintaining the CSS side of things,
but adding back the secondary serialization of the color Style types. To make
things work, the new secondary serialization functions all needed a new name.
`serializationForCSSTokenization` was chosen to clarify these are really only
for the tokenization needs.

* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-006-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/registered-property-computation-color-006.html: Added.
* Source/WebCore/style/StyleCustomProperty.cpp:
* Source/WebCore/style/StyleCustomProperty.h:
* Source/WebCore/style/values/color/StyleColor.cpp:
* Source/WebCore/style/values/color/StyleColor.h:
* Source/WebCore/style/values/color/StyleColorLayers.cpp:
* Source/WebCore/style/values/color/StyleColorLayers.h:
* Source/WebCore/style/values/color/StyleColorMix.cpp:
* Source/WebCore/style/values/color/StyleColorMix.h:
* Source/WebCore/style/values/color/StyleContrastColor.cpp:
* Source/WebCore/style/values/color/StyleContrastColor.h:
* Source/WebCore/style/values/color/StyleCurrentColor.cpp:
* Source/WebCore/style/values/color/StyleCurrentColor.h:
* Source/WebCore/style/values/color/StyleRelativeColor.h:
* Source/WebCore/style/values/color/StyleResolvedColor.cpp:
* Source/WebCore/style/values/color/StyleResolvedColor.h:

Canonical link: https://commits.webkit.org/297600@main
